speech/quickstart.v2.js

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function main(
+  projectId,
+  recognizerId = 'my-recognizer',
+  audioFilePath = 'resources/brooklyn.flac'
+) {
+  // [START speech_quickstart_v2]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = "your-project-id";
+  // const recognizerId = "your-recognizer-id";
+  // const audioFilePath = "path/to/audio/file";
+
+  // Imports the Google Cloud Speech-to-Text library (v2)
+  const speech = require('@google-cloud/speech');
+  const fs = require('fs');
+  async function quickstartV2() {
+    // Instantiates a client
+    const client = new speech.v2.SpeechClient();
+
+    const recognizerRequest = {
+      parent: `projects/${projectId}/locations/global`,
+      recognizerId: recognizerId,
+      recognizer: {
+        languageCodes: ['en-US'],
+        model: 'latest_long',
+      },
+    };
+
+    const operation = await client.createRecognizer(recognizerRequest);
+    const recognizer = operation[0].result;
+
+    const content = fs.readFileSync(audioFilePath).toString('base64');
+
+    const transcriptionRequest = {
+      recognizer: recognizer.name,
+      config: {
+        // Automatically detects audio encoding
+        autoDecodingConfig: {},
+      },
+      content: content,
+    };
+
+    const response = await client.recognize(transcriptionRequest);
+    for (const result of response[0].results) {
+      console.log(`Transcript: ${result.alternatives[0].transcript}`);
+    }
+  }
+
+  quickstartV2();
+  // [END speech_quickstart_v2]
+}
+
+process.on('unhandledRejection', err => {
+  console.error(err.message);
+  process.exitCode = 1;
+});
+main(...process.argv.slice(2));
```

Comment:
The [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/#no-cli) recommends avoiding CLIs.

**For testing**: can we replace this with [exporting `main`](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/2886/files#diff-a37495ea38a7e938e30723870b6c8a6a2e966a6305429628f1e435b4c0cf3a05R51) call and [`console.log` stubbing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/2886/files#diff-dc9e6f35710dbb4c358e05f0bb9218021f03b8fc4db9a27bf0f91d6255b147e3R305)?

_Apply this comment throughout your PR._

Reply:
Thank you for the review! I've removed the CLI stuff and used stubbing instead.